src/Messaging/test/RabbitMQ.Test/Host/RabbitMQHostWebApplicationTest.cs

```diff
@@ -0,0 +1,135 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+using Steeltoe.Common.Lifecycle;
+using Steeltoe.Connector.RabbitMQ;
+using Steeltoe.Extensions.Configuration.CloudFoundry;
+using Steeltoe.Messaging.RabbitMQ.Configuration;
+using Xunit;
+using RC = RabbitMQ.Client;
+
+namespace Steeltoe.Messaging.RabbitMQ.Host;
+
+public class RabbitMQHostWebApplicationTest
+{
+    [Fact]
+    public void WebAppHostCanBeStarted()
+    {
+        MockRabbitHostedService hostedService;
+
+        WebApplicationBuilder builder = RabbitMQHost.CreateWebApplicationBuilder(null);
+        ;
+        builder.Services.AddSingleton<IHostedService, MockRabbitHostedService>();
+
+        using (WebApplication webApp = builder.Build())
+        {
+            webApp.Start();
+            hostedService = webApp.Services.GetServices<IHostedService>().OfType<MockRabbitHostedService>().First();
+            Assert.NotNull(hostedService);
+            Assert.Equal(1, hostedService.StartCount);
+            Assert.Equal(0, hostedService.StopCount);
+            Assert.Equal(0, hostedService.DisposeCount);
+            webApp.StopAsync().Wait();
+        }
+
+        Assert.Equal(1, hostedService.StartCount);
+        Assert.Equal(1, hostedService.StopCount);
+        Assert.Equal(1, hostedService.DisposeCount);
+    }
+
+    [Fact]
+    public void WebAppHostShouldInitializeServices()
+    {
+        using WebApplication webApp = RabbitMQHost.CreateWebApplicationBuilder().Build();
+        webApp.Start();
+        var lifecycleProcessor = webApp.Services.GetRequiredService<ILifecycleProcessor>();
+        RabbitHostService rabbitHostService = webApp.Services.GetServices<IHostedService>().OfType<RabbitHostService>().First();
+
+        Assert.True(lifecycleProcessor.IsRunning);
+        Assert.NotNull(rabbitHostService);
+        webApp.StopAsync();
+    }
+
+    [Fact]
+    public void WebAppHostShouldAddRabbitOptionsConfiguration()
+    {
+        WebApplicationBuilder builder = RabbitMQHost.CreateWebApplicationBuilder();
+
+        var appSettings = new Dictionary<string, string>
+        {
+            [$"{RabbitOptions.Prefix}:host"] = "ThisIsATest",
+            [$"{RabbitOptions.Prefix}:port"] = "1234",
+            [$"{RabbitOptions.Prefix}:username"] = "TestUser",
+            [$"{RabbitOptions.Prefix}:password"] = "TestPassword"
```

Comment:
If I understand correctly, this test fails because:

- `RabbitOptions.Prefix` is `spring:rabbitmq`, but Connectors use `rabbitmq:client:` or `rabbit:client`
- Port is never set [here](https://github.com/SteeltoeOSS/Steeltoe/blob/main/src/Messaging/src/RabbitMQ/Extensions/RabbitServicesExtensions.cs#L238)
- `host` is referred to as `server` in Connectors
- The hostname is converted to lowercase when parsed as a Uri somewhere in connectors